### PR TITLE
Updates needed for clean autodoc build with doxygen-1.8.17.

### PR DESCRIPTION
**Pull request by KineticTheory**
### Background

+ While debugging an autodoc build error related to a PR, I decided to try a newer version of doxygen.  Previously we were using 1.8.14/15 and these updates are needed to support version 1.8.17.

### Description of changes

+ The `doxygen_config` has changed a bit and extra logic was added to the build system to support doxygen version 1.8.14 through 1.8.17.  These changes are related to the doxygen configure options `PERL_PATH` and `DOXYGEN_HTML_DYNAMIC_MENUS`.
+ I removed `$prefix/install` from doxygen's inlcude path for now because it was generating a warning.  I may need to add it back with more complicated logic to allow client codes to build autodoc.
+ The new version of doxygen is pickier about missing `\ref` commands and duplicate entries.  I updated file content to allow a clean autodoc build. Most changes were minor.  If `@param` arguments were listed in both `.hh` and `.cc` files, I removed the duplicates from the `.hh` file.  In some `.dcc` files, I removed `\ref` entries.  In some cases I removed large blocks of text that were no longer valid (autoconf build options, etc.).

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [x] Travis/Appveyor CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Toss3 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Trinitite checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

**Comment by KineticTheory** [comment]
@keadyk Can you take a look at these changes?  They need to be merged so we can take a closer look at the failing autodoc CI for your MR.

**Comment by keadyk** [comment]
On it! :)

File: config/doxygen_config.in [hunk 1]
```diff
@@ -2095,7 +2095,7 @@ INCLUDE_FILE_PATTERNS  =
 # recursively expanded use the := operator instead of the = operator.
 # This tag requires that the tag ENABLE_PREPROCESSING is set to YES.
 
-PREDEFINED             =
+PREDEFINED             = "Remember(c)=c"
```
Comment (keadyk) [inline config/doxygen_config.in:2098]: So this just replaces the text of the Remember() with the variable itself? :)

File: src/RTT_Format_Reader/autodoc/RTT_Format_Reader.dcc.in [hunk 2]
```diff
@@ -273,12 +270,12 @@ General comments:
 
 \page rtt_stdcell RTT Format ICEM/DDN Cell Definitions
 
-The RTT_Formatt_Reader side set numbering output by ICEM/DDN is
-depicted on this page. Note that the "right hand rule" is used to
-return the direction of the outward-directed normal when the nodes are
-traversed in the order that is specified in the side set node
-ordering. The RTT_Format cell definitions do not assume any particular
-orientation of the sides relative to the problem coordinate system.
+The RTT_Formatt_Reader side set numbering output by ICEM/DDN is depicted on this
```
Comment (keadyk) [inline src/RTT_Format_Reader/autodoc/RTT_Format_Reader.dcc.in:273]: Formatt->Format :)

File: config/doxygen_config.in [hunk 3]
```diff
@@ -2166,7 +2166,8 @@ EXTERNAL_PAGES         = YES
 # interpreter (i.e. the result of 'which perl').
 # The default file (with absolute path) is: /usr/bin/perl.
 
-PERL_PATH              = /usr/bin/perl
+# PERL_PATH              = /usr/bin/perl
+@PERL_PATH@
```
Comment (keadyk) [inline config/doxygen_config.in:2170]: Is this change related to spack, or is there some other reason we might expect the PERL_PATH to be something other than usr/bin/perl?

Comment (KineticTheory) [reply config/doxygen_config.in:2170]: The `PERL_PATH` input is deprecated by newer doxygen versions.  This entry causes an error if left in place.

File: config/doxygen_config.in (hunk 1 above)
Comment (KineticTheory) [reply config/doxygen_config.in:2098]: This tells doxygen to always replace `Remember(c)` with `c` and is independent of the `DBC_LEVEL` setting.

File: src/RTT_Format_Reader/autodoc/RTT_Format_Reader.dcc.in (hunk 2 above)
Comment (keadyk) [reply src/RTT_Format_Reader/autodoc/RTT_Format_Reader.dcc.in:273]: I'll make this change with the GitHub editor and merge

**Comment by keadyk** [comment]
@KineticTheory I'll squash and merge once the GitHub checks come back, if that is okay with you! I made the spelling change.

**Pull request merged**